year2024/Day1.java

```diff
@@ -0,0 +1,1062 @@
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Day1 {
+    
+    public static void main(String[] args) {
+        // initialize
+        String[] twoLocationIds = input.split("\n");
+        int[] aLocationIds = new int[twoLocationIds.length];
+        fillLocationIds(twoLocationIds, aLocationIds, 0);
+        int[] bLocationIds = new int[twoLocationIds.length];
+        fillLocationIds(twoLocationIds, bLocationIds, 1);
+        
+        // Part1
+        calculateDistance(aLocationIds, bLocationIds);
+        // Part2
+        calculateSimilarityScore(aLocationIds, bLocationIds);
+    }
+    
+    // initialize
+    private static void fillLocationIds(String[] twoLocationIds, int[] locationIds, int index) {
+        for (int i = 0; i < locationIds.length; i++) {
+            String[] ids = twoLocationIds[i].split("   ");
```

Comment:
このコードは `"   "`（スペース3つ分）のときのみしか動かないので、スペースの数に限らず動くようにしたほうがよいと思います。

以下のようにした場合、前後のスペースや数字の間のスペースの数に関わらず欲しい出力が得られそうです。たとえば `"   1 2  3   4   "` が与えられても `["1", "2", "3", "4"]` が得られるはずです。（実行確認はしていないので間違っているかもですが）

```suggestion
            String[] ids = twoLocationIds[i].trim().split("\\s+");
```

Reply:
ありがとうございます。
簡潔で変化にも対応できて勉強になりました！
Step4 で追加しました。